client/Routes.js

```diff
@@ -1,40 +1,72 @@
 import React, { Component, Fragment } from 'react';
 import { connect } from 'react-redux';
 import { withRouter, Route, Switch, Redirect } from 'react-router-dom';
-import { Login, Signup } from './components/AuthForm';
+import { Login, Signup } from './components/AuthForm.js';
 import Home from './components/Home';
 import AllProducts from './components/AllProducts';
 import SingleProduct from './components/SingleProduct';
+import Brands from './components/Brands';
+import OrderHistory from './components/OrderHistory';
+import ProductReviews from './components/ProductReviews';
+import CreateReview from './components/CreateReview';
+import UpdateProduct from './components/UpdateProduct';
+import Cart from './components/Cart';
+import NotFoundPage from './components/NotFoundPage';
 import { me } from './store';
+import { fetchProducts } from "./store/products";
+import { fetchCart } from "./store/cart";
 
 /**
  * COMPONENT
  */
 class Routes extends Component {
   componentDidMount() {
     this.props.loadInitialData();
+    this.props.fetchProducts();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (!prevProps.isLoggedIn && this.props.isLoggedIn) {
+      this.props.fetchCart();
+    }
   }
 
   render() {
     const { isLoggedIn } = this.props;
 
     return (
       <div>
-        {isLoggedIn ? (
-          <Switch>
-            <Route path="/" component={Home} />
-            <Redirect to="/home" />
-            <Route exact path="/products" component={AllProducts} />
-            <Route exact path="/products/:id" component={SingleProduct} />
-          </Switch>
-        ) : (
-          <Switch>
-            <Route path="/login" component={Login} />
-            <Route path="/signup" component={Signup} />
-            <Route exact path="/products" component={AllProducts} />
-            <Route exact path="/products/:id" component={SingleProduct} />
-          </Switch>
-        )}
+        <Switch>
+          {/* This is the ROUTES for ALL visitors */}
+          <Route path='/login' component={Login} />
+          <Route path='/signup' component={Signup} />
+          <Route path='/home' component={AllProducts} />
+          <Route path='/cart' component={Cart} />
+          <Route exact path='/products' component={AllProducts} />
+          <Route path='/products/:id' component={SingleProduct} />
+          <Route exact path='/brands' component={Brands} />
+          <Route path='/brands/:id' component={SingleProduct} />
+          <Route path='/products/:id/update' component={UpdateProduct} />
+          <Route path='/products/:id/reviews' component={ProductReviews} />
+          <Route path='/products/:id/add' component={CreateReview} />
+          <Route component={NotFoundPage} />
+          <Redirect from='/login' to='/home' />
+
+          {isLoggedIn && (
+            // These are the routes available for the users LOGGED IN
+            <Switch>
```

Comment:
These individual routes should each be guarded by an `isLoggedIn &&`, rather than having a new switch statement nested here. You also shouldn't need many, as lots of these routes are already registered for all users above